pkg/controller/import-controller.go

```diff
@@ -803,6 +807,76 @@ func (r *ImportReconciler) getVddkImageName() (*string, error) {
 	return nil, errors.Errorf("found %s ConfigMap in namespace %s, but it does not contain a '%s' entry", common.VddkConfigMap, namespace, common.VddkConfigDataKey)
 }
 
+func (r *ImportReconciler) updatePVCBoundContion(pvc *corev1.PersistentVolumeClaim) error {
+	// set bound condition by getting the latest event
+	events := &corev1.EventList{}
+
+	err := r.client.List(context.TODO(), events,
+		client.InNamespace(pvc.GetNamespace()),
+		client.MatchingFields{"involvedObject.name": pvc.GetName(),
+			"involvedObject.uid": string(pvc.GetUID())},
+	)
+
+	if err != nil {
+		// Log the error but don't fail the reconciliation
+		r.log.Error(err, "Unable to list events for PVC bound condition update", "pvc", pvc.Name)
+		return nil
+	}
+
+	if len(events.Items) == 0 {
+		return nil
+	}
+
+	pvcPrime, exists := pvc.GetAnnotations()[cc.AnnPVCPrimeName]
+
+	// Sort event lists by containing primeName substring and most recent timestamp
+	sort.Slice(events.Items, func(i, j int) bool {
+		if exists {
+			firstContainsPrime := strings.Contains(events.Items[i].Message, pvcPrime)
+			secondContainsPrime := strings.Contains(events.Items[j].Message, pvcPrime)
+
+			if firstContainsPrime && !secondContainsPrime {
+				return true
+			}
+			if !firstContainsPrime && secondContainsPrime {
+				return false
+			}
+		}
+		// if both contains primeName substring or neither, just sort on timestamp
+		return events.Items[i].FirstTimestamp.Time.After(events.Items[j].FirstTimestamp.Time)
+	})
+
+	boundMessage := ""
+
+	if exists {
+		// if we are using populators get the latest event from prime pvc
+		pvcPrime = fmt.Sprintf("[%s] : ", pvcPrime)
+
+		// split so we can remove prime name prefix from event message
+		res := strings.Split(events.Items[0].Message, pvcPrime)
+		boundMessage = res[len(res)-1]
+
+		if boundMessage == "" {
+			return nil
+		}
+	} else {
+		// if not using populators just get the latest event
+		boundMessage = events.Items[0].Message
+	}
+
+	anno := pvc.GetAnnotations()
```

Comment:
Seems there's a panic in this function (check [this log](https://storage.googleapis.com/kubevirt-prow/pr-logs/pull/kubevirt_containerized-data-importer/3764/pull-containerized-data-importer-e2e-istio/1938001112250454016/artifacts/1/pods/1_cdi-zdxo7_cdi-deployment-7c9cfc486d-n5hhm-cdi-deployment.log)). We should probably nil check these annotations in case they are nil.